src/pkg/clients/Amqp.go

```diff
@@ -1,39 +1,74 @@
 package clients
 
 import (
+	"time"
+
+	log "github.com/sirupsen/logrus"
 	"github.com/streadway/amqp"
 )
 
 type IAmqpClient interface {
+	WithRedial() chan IAmqpClient
+	Close() error
 	Publish(nameQueue string, message []byte) error
 	Consume(nameQueue string) (<-chan amqp.Delivery, error)
-	QueueDeclare() (amqp.Queue, error)
 }
 
 var _ IAmqpClient = &amqpClient{}
 
 type amqpClient struct {
-	channel   *amqp.Channel
-	fullAddr  string
-	queueName string
+	connection *amqp.Connection
+	channel    *amqp.Channel
+	address    string
 }
 
-func NewAmqpClient(addr, user, pwd, queueName string) (IAmqpClient, error) {
+func NewAmqpClient(address string) (IAmqpClient, error) {
```

Comment:
Why did we chose to deport the address construction outside of the client Constructor ? The string building code is now duplicated if I'm correct.